### PR TITLE
Factor out `confirm`, `placeholder`, `focus_on_load` and `dispatch_action_config` properties

### PR DESCRIPTION
**Pull request by filmaj**
The idea behind this PR is to have a set of interfaces that other Block Kit elements can extend from easily.

This PR also starts adding some of the documentation hints from api.slack.com as JSDoc comments.

It is a small refactor, but it is especially helpful when documenting properties, as then they can be documented in one place but used in many places.

Let me know what you think! Like/dislike/don't care?

### CODE REVIEW

File: packages/types/src/index.ts [hunk 1]
```diff
@@ -121,181 +137,170 @@ export type MultiSelect =
...
-export interface MultiUsersSelect extends Action {
+export interface Placeholderable {
+  /**
+   * @description A {@see PlainTextElement} object that defines the placeholder text shown on the element. Maximum
+   * length for the `text` field in this object is 150 characters.
+   */
+  placeholder?: PlainTextElement;
+}
+
+export interface Dispatchable {
```
Comment (filmaj) [inline packages/types/src/index.ts:173]: Right now only one element uses this interface, but if/when we merge https://github.com/slackapi/node-slack-sdk/pull/1549, then the new input elements like URL, email and number inputs will also use this interface.

File: packages/types/src/index.ts [hunk 2]
```diff
@@ -121,181 +137,170 @@ export type MultiSelect =
...
+  /**
+   * @description Indicates whether the element will be set to auto focus within the
+   * {@link https://api.slack.com/reference/surfaces/views `view` object}. Only one element can be set to `true`.
+   * Defaults to false.
+   */
   focus_on_load?: boolean;
 }
 
-export interface MultiUsersSelect extends Action {
+export interface Placeholderable {
```
Comment (srajiang) [inline packages/types/src/index.ts:165]:
```suggestion
export interface Placeholdable {
```
Too many syllables for my little brain

File: packages/types/src/index.ts [hunk 3]
```diff
@@ -121,181 +137,170 @@ export type MultiSelect =
...
+
+export interface Dispatchable {
+  /**
+   * @description A {@see DispatchActionConfig} object that determines when during text input the element returns a
+   * {@link https://api.slack.com/reference/interaction-payloads/block-actions `block_actions` payload}.
+   */
+  dispatch_action_config?: DispatchActionConfig;
+}
+
+export interface UsersSelect extends Action, Confirmable, Focusable, Placeholderable {
```
Comment (srajiang) [inline packages/types/src/index.ts:181]: :chefskiss:

**Review by WilliamBergamin — APPROVED** [review]
Great work 💯 this reminds me of a decorator pattern

**Review by srajiang — CHANGES_REQUESTED** [review]
I like this direction, and thank you for additional jsdoc to help document things. Left a minor comment about naming.

File: packages/types/src/index.ts [hunk 4]
```diff
@@ -121,181 +137,170 @@ export type MultiSelect =
...
+   * with.
+   */
   confirm?: Confirm;
+}
+
+export interface Focusable {
+  /**
+   * @description Indicates whether the element will be set to auto focus within the
+   * {@link https://api.slack.com/reference/surfaces/views `view` object}. Only one element can be set to `true`.
+   * Defaults to false.
```
Comment (mwbrooks) [inline packages/types/src/index.ts:160]:
```suggestion
   * Defaults to `false`.
```

**Review by mwbrooks — APPROVED** [review]
<3 Love it! Only found a minor improvement suggestion.

**Review by e-zim — APPROVED** [review]
These documentation hints are wonderful, particularly when `@see` comes into play! I'm a huge fan 👏

**Pull request merged**